### PR TITLE
Send mail order with background job 

### PR DESCRIPTION
**Pull request by TanBinh1601**
## Related Tickets
- [#44906](https://edu-redmine.sun-asterisk.vn/issues/44906)

## WHAT (optional)
- Handle sending email in background job
- Statistics of email sending activity for users

## HOW
- gem "sidekiq "

## Evidence (Screenshot or Video)
Screenshot from 2022-01-07 12-06-53
Screenshot from 2022-01-07 12-04-19

### CODE REVIEW

**Comment by TanBinh1601** [comment]
ready

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -90,7 +90,6 @@ def logged_in_user
   end
 
   def send_mail_new_order
-    OrderMailer.new_orders(@order, @order.order_details.includes(:product),
-                           @order.total_money).deliver_now
+    SendEmailJob.set(wait: 1.minute).perform_later @order
```
Comment (huongnt-2545) [inline app/controllers/orders_controller.rb]: `1` cho vào settings em nhé

File: app/views/order_mailer/new_orders.html.erb [hunk 2]
```diff
@@ -13,7 +13,7 @@
     </tr>
   </thead>
   <tbody>
-    <% @order_detail.each do |detail|%>
+    <% @order.order_details.includes(:product).each do |detail|%>
```
Comment (huongnt-2545) [inline app/views/order_mailer/new_orders.html.erb]: đưa vào controller để xử lý em, ko tương tác lấy data từ view như này

Comment (TanBinh1601) [reply app/views/order_mailer/new_orders.html.erb]: Vâng ạ...

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (TanBinh1601) [reply app/controllers/orders_controller.rb]: dạ...

**Comment by TanBinh1601** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**